src/Model/PropertyGo.cs

```diff
@@ -13,60 +12,82 @@ public class PropertyGo : Property
     {
         public PropertyGo()
         {
-
         }
 
-        public string Tag()
+        public string Tag(bool omitEmpty = true)
         {
             // don't emit a tag if this property is part of a parameter group (it's not necessary)
             if (Extensions.ContainsKey(SwaggerExtensions.ParameterGroupExtension))
             {
                 return string.Empty;
             }
 
-            if (Parent.CodeModel.ShouldGenerateXmlSerialization)
+            if (!Parent.CodeModel.ShouldGenerateXmlSerialization)
             {
-                var sb = new StringBuilder("`xml:\"");
+                return string.Format("`json:\"{0}{1}\"`", SerializedName, omitEmpty ? ",omitempty" : "");
+            }
 
-                bool hasParent = false;
-                if (Parent is CompositeTypeGo && !((CompositeTypeGo)Parent).IsWrapperType)
-                {
-                    sb.Append(XmlName);
-                    hasParent = true;
-                }
+            var sb = new StringBuilder("`xml:\"");
 
-                if (XmlIsWrapped)
-                {
-                    if (hasParent)
-                    {
-                        sb.Append('>');
-                    }
+            bool hasParent = false;
+            if (Parent is CompositeTypeGo go && !go.IsWrapperType)
+            {
+                sb.Append(SerializedName);
+                hasParent = true;
+            }
 
-                    var asSequence = ModelType as SequenceTypeGo;
-                    sb.Append(asSequence.ElementXmlName);
-                }
-                else if (XmlIsAttribute)
+            if (XmlIsWrapped)
+            {
+                if (hasParent)
                 {
-                    sb.Append(",attr");
+                    sb.Append('>');
                 }
 
-                sb.Append("\"`");
-                return sb.ToString();
+                var asSequence = ModelType as SequenceTypeGo;
+                sb.Append(asSequence.ElementXmlName);
             }
+            else if (XmlIsAttribute)
+            {
+                sb.Append(",attr");
+            }
+
+            sb.Append("\"`");
+            return sb.ToString();
 
-            return string.Format("`json:\"{0},omitempty\"`", SerializedName);
         }
 
+        /// <summary>
+        ///  Gets if the property should be treated as a pointer
+        /// </summary>
+        public bool IsPointer => !(this.ModelType.HasInterface() || IsRequired || ModelType.CanBeNull() || ModelType is EnumTypeGo);

```

Comment:
>EnumTypeGo [](start = 120, length = 10)

is v2 we had the extra condition here to be a named enum to be a value type. But I saw we were not using that in v3. Let me know if we want to add the extra criteria